### PR TITLE
Add reftests for trigonometric functions

### PR DESCRIPTION
**Pull request by gagern**
This continues #227 by adding some more tests, and generally extending the documentation.

### CODE REVIEW

File: ref/Arithmetic_Operators.md [hunk 1]
```diff
@@ -239,37 +239,201 @@ Complex numbers are fully supported.
 
 The standard trigonometric functions are available through the following operators:
 
-#### Trigonometric sine function: `sin(‹expr1›)`
+#### Trigonometric sine function: `sin(‹expr›)`
 
-#### Trigonometric cosine function: `cos(‹expr1›)`
+The argument is in an angle, given in radians.
```
Comment (kranich) [inline ref/Arithmetic_Operators.md]: The argument is ~~in~~ an angle, given in radians.

File: ref/Arithmetic_Operators.md [hunk 2]
```diff
@@ -239,37 +239,201 @@ Complex numbers are fully supported.
...
+    > sin(90) // Don't forget the degree sign if your angle is in degree!
+    < 0.894
+    > sin(2 + 3*i)
+    < 9.1545 - i*4.1689
 
-#### Inverse trigonometric sine function: `arcsin(‹expr1›)`
+#### Trigonometric cosine function: `cos(‹expr›)`
 
-#### Inverse trigonometric cosine function: `arccos(‹expr1›)`
+The argument is in an angle, given in radians.
```
Comment (kranich) [inline ref/Arithmetic_Operators.md]: Same as above.

File: ref/Arithmetic_Operators.md [hunk 3]
```diff
@@ -239,37 +239,201 @@ Complex numbers are fully supported.
...
+    > cos(90) // Don't forget the degree sign if your angle is in degree!
+    < -0.4481
+    > cos(2 + 3*i)
+    < -4.1896 - i*9.1092
 
-#### Inverse rigonometric tangent function: `arctan(‹expr1›)`
+#### Trigonometric tangent function: `tan(‹expr›)`
 
-#### Angle of a vector: `arctan2(‹expr1,expr2›)`
+The argument is in an angle, given in radians.
```
Comment (kranich) [inline ref/Arithmetic_Operators.md]: Same as above.

File: ref/Arithmetic_Operators.md [hunk 4]
```diff
@@ -239,37 +239,201 @@ Complex numbers are fully supported.
...
+    < -3.0186 + i*4.567
+    > arctan2(r*cos(a), r*sin(a)) + pi
+    < 0.123 + i*4.567
+
+Since real arguments and complex arguments follow different code paths,
+it is important to verify that a small complex perturbation doesn't lead
+to different branch choices.
+
+    > jitter = i * 1.234 * 10^(-7);
+    > select(directproduct([-3, 3], [-4, 4]), // try all quadrants
```
Comment (kranich) [inline ref/Arithmetic_Operators.md]: Much nicer than my unit test!

**Comment by kranich** [comment]
@gagern Very nice documentation, thank you!

**Comment by gagern** [comment]
Thanks for the review! I was tempted to let this pass as a reviewer comment on your pull request, but it was a good thing you had a look at this and spotted my mistake.

**Pull request merged**